### PR TITLE
nMFI normalisation type

### PR DESCRIPTION
**Pull request by Fersoil**
- Added function `get_nmfi`
- modified `process_plate` function to handle new normalisation type
- added tests and docs for the `get_nMFI` function
- added tests for `process_plate`

This PR should be merged after a successful review of #171

### CODE REVIEW

**Review by ZetrextJG — CHANGES_REQUESTED** [review]
Nice work. I like that the `get_nmfi` function is kinda separated. I would consider extending the vignettes and state there clearly what nMFI is and how can a user get that data.

Besides that i have two minor nitpicks into the code, not a big deal, but I would like to

File: tests/testthat/test-get-nmfi.R
[the remark this answers is not in the document]
Comment (Fersoil) [reply tests/testthat/test-get-nmfi.R:1]: done

File: R/process-plate.R [hunk 1]
```diff
@@ -29,34 +62,74 @@
...
+        output_path,
+        "'\n",
+        verbose = verbose
+      )
+      write.csv(nmfi, output_path, row.names = FALSE)
+      return(nmfi)
+    }
+
+
+    # RAU normalisation
```
[the remark this answers is not in the document]
Comment (Fersoil) [reply R/process-plate.R:110]: added the `else if` block, the data format is validated above (line `    stopifnot(is_valid_normalisation_type(normalisation_type))
`)

[the remark this answers is not in the document]
Comment (Fersoil) [reply R/process-plate.R:110]: will that be enough?

[the remark this answers is not in the document]
Comment (ZetrextJG) [reply R/process-plate.R]: Yeah that is enough

File: vignettes/example_script.Rmd [hunk 2]
```diff
@@ -207,5 +219,18 @@ plot_standard_curve_analyte_with_model(plate, model, log_scale = c("all"))
 plot_standard_curve_analyte_with_model(plate, model, log_scale = c("all"), over_max_extrapolation = 100000)
 ```
 
+### nMFI
+In some cases, the RAU values cannot be properly calculated. It may happen, for instance when the MFI values of tests samples are way higher than the standard curve samples. In that case, to avoid extrapolation, but to be still able to compare the samples across the plates, we introduced a new unit called nMFI (Normalized MFI). The nMFI is calculated as the MFI value of the test sample divided by the MFI value of the standard curve sample with selected dilution value.
```
Comment (ZetrextJG) [inline vignettes/example_script.Rmd:223]: `
```suggestion
In some cases, the RAU values cannot be reliably calculated. It may happen, for instance when the MFI values of tests samples are way higher than the standard curve samples. In that case, to avoid extrapolation, but to be still able to compare the samples across the plates, we introduced a new unit called nMFI (Normalized MFI). The nMFI is calculated as the MFI value of the test sample divided by the MFI value of the standard curve sample with selected dilution value.
```

File: R/get-nmfi.R [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+           " dilution is not present in the plate.")
+    }
+
+
+    # get index of standard curve sample with the target dilution
+    reference_standard_curve_name <-
+      subset(plate$sample_names,
+             plate$dilution_values == reference_dilution)
+    reference_standard_curve_id <-
+      which(plate$sample_names == reference_standard_curve_name)
```
Comment (ZetrextJG) [inline R/get-nmfi.R:84]: The `sample_names` and `dilution_values` are by design aligned. So a more reliable way of getting that value is
```r
reference_standard_curve_id <- which(
  plate$dilution_values == reference_dilution
)
stopifnot(length(reference_standard_curve_id) == 1)
```

File: R/get-nmfi.R [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    plate_data <- plate$data[[data_type]]
```
Comment (ZetrextJG) [inline R/get-nmfi.R:86]: You should use the `get_data` method as it handles edge cases better

File: R/get-nmfi.R [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+    reference_mfi <- plate_data[reference_standard_curve_id,]
+
+    test_mfi <- plate_data[plate$sample_types == "TEST",]
```
Comment (ZetrextJG) [inline R/get-nmfi.R:90]: Same here, use `get_data`

**Review by ZetrextJG — CHANGES_REQUESTED** [review]
Nice to see vignettes. I added some code concerns that I have not caught in the first round of review

**Review by ZetrextJG — APPROVED** [review]
LGTM
